tests/framework/e2e/cluster.go

```diff
@@ -554,7 +554,9 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
 		args = append(args, "--experimental-warning-unary-request-duration", cfg.ExperimentalWarningUnaryRequestDuration.String())
 	}
 	if cfg.SnapshotCatchUpEntries > 0 {
-		args = append(args, "--experimental-snapshot-catchup-entries", fmt.Sprintf("%d", cfg.SnapshotCatchUpEntries))
+		if cfg.Version == CurrentVersion || (cfg.Version == MinorityLastVersion && i <= cfg.ClusterSize/2) || (cfg.Version == QuorumLastVersion && i > cfg.ClusterSize/2) {
+			args = append(args, "--experimental-snapshot-catchup-entries", fmt.Sprintf("%d", cfg.SnapshotCatchUpEntries))
+		}
 	}
```

Comment:
Good catch!

It's a test bug introduced in https://github.com/etcd-io/etcd/pull/15033. Since the flag `--experimental-snapshot-catchup-entries` is supported starting from 3.6.0 (`main` branch for now), so we should ignore the flag if the etcd binary is on an old version, e.g. 3.5.6.

The suggested fix is below. 
cc @chaochn47 

```
$ git diff
diff --git a/tests/framework/e2e/cluster.go b/tests/framework/e2e/cluster.go
index 870dc86de..ceca85326 100644
--- a/tests/framework/e2e/cluster.go
+++ b/tests/framework/e2e/cluster.go
@@ -513,6 +513,28 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
                args = append(args, "--listen-metrics-urls", murl)
        }
 
+       var execPath string
+       switch cfg.Version {
+       case CurrentVersion:
+               execPath = BinPath.Etcd
+       case MinorityLastVersion:
+               if i <= cfg.ClusterSize/2 {
+                       execPath = BinPath.Etcd
+               } else {
+                       execPath = BinPath.EtcdLastRelease
+               }
+       case QuorumLastVersion:
+               if i <= cfg.ClusterSize/2 {
+                       execPath = BinPath.EtcdLastRelease
+               } else {
+                       execPath = BinPath.Etcd
+               }
+       case LastVersion:
+               execPath = BinPath.EtcdLastRelease
+       default:
+               panic(fmt.Sprintf("Unknown cluster version %v", cfg.Version))
+       }
+
        args = append(args, cfg.TlsArgs()...)
 
        if cfg.AuthTokenOpts != "" {
@@ -553,7 +575,9 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
        if cfg.ExperimentalWarningUnaryRequestDuration != 0 {
                args = append(args, "--experimental-warning-unary-request-duration", cfg.ExperimentalWarningUnaryRequestDuration.String())
        }
-       if cfg.SnapshotCatchUpEntries > 0 {
+       // TODO: remove `&& execPath == BinPath.Etcd ` after we cut the release-3.6 branch.
+       // Note `--experimental-snapshot-catchup-entries` is supported starting from 3.6.0.
+       if cfg.SnapshotCatchUpEntries > 0 && execPath == BinPath.Etcd {
                args = append(args, "--experimental-snapshot-catchup-entries", fmt.Sprintf("%d", cfg.SnapshotCatchUpEntries))
        }
        envVars := map[string]string{}
@@ -566,28 +590,6 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
                envVars["GOFAIL_HTTP"] = fmt.Sprintf("127.0.0.1:%d", gofailPort)
        }
 
-       var execPath string
-       switch cfg.Version {
-       case CurrentVersion:
-               execPath = BinPath.Etcd
-       case MinorityLastVersion:
-               if i <= cfg.ClusterSize/2 {
-                       execPath = BinPath.Etcd
-               } else {
-                       execPath = BinPath.EtcdLastRelease
-               }
-       case QuorumLastVersion:
-               if i <= cfg.ClusterSize/2 {
-                       execPath = BinPath.EtcdLastRelease
-               } else {
-                       execPath = BinPath.Etcd
-               }
-       case LastVersion:
-               execPath = BinPath.EtcdLastRelease
-       default:
-               panic(fmt.Sprintf("Unknown cluster version %v", cfg.Version))
-       }
-
        return &EtcdServerProcessConfig{
                lg:           cfg.Logger,
                ExecPath:     execPath,
```

Reply:
Make sense to me. Good catch!